### ISSUE TITLE
[REVIEW]: DisjunctiveProgramming.jl: Generalized Disjunctive Programming Models and Algorithms for JuMP

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#117]
**Submitting author:** @hdavid16 (<a href="http://orcid.org/0000-0002-7374-1120">Hector Perez</a>)
**Repository:** <a href="https://github.com/hdavid16/DisjunctiveProgramming.jl" target ="_blank">https://github.com/hdavid16/DisjunctiveProgramming.jl</a>
**Version:** v0.4.1
**Reviewers:** @sbolusani, @joaquimg, @joaquimg
**Archive:** <a href="https://doi.org/10.5281/zenodo.10144853" target="_blank">10.5281/zenodo.10144853</a>

## Status

[![status](https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c/status.svg)](https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c)

Status badge code:

```
HTML: <a href="https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c"><img src="https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c/status.svg"></a>
Markdown: [![status](https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c/status.svg)](https://proceedings.juliacon.org/papers/569beb65dae7d35e1ccf220df54e8a1c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/hdavid16/DisjunctiveProgramming.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@sbolusani & @joaquimg, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @matbesancon know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/hdavid16/DisjunctiveProgramming.jl">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Authorship:** Has the submitting author (@hdavid16) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [ ] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [ ] **Context:** is the scientific context motivating the work correctly presented?
- [ ] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [ ] **Results:** are the results presented and compared to approaches with similar goals?
### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/hdavid16/DisjunctiveProgramming.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@hdavid16) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Wordcount for `paper.tex` is 3421

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.09 s (406.4 files/s, 220866.5 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
CSS                              2            653             33          14577
TeX                              8            282            178           2855
Julia                           13            149            115           1009
JavaScript                       4             38             64            450
Markdown                         2             20              0             71
YAML                             4              2              0             71
Ruby                             1              8              4             45
TOML                             2              4              0             19
HTML                             2              0              0              4
-------------------------------------------------------------------------------
SUM:                            38           1156            394          19101
-------------------------------------------------------------------------------


Statistical information for the repository '4bdf66050d86061e21566b57' was
gathered on 2022/12/11.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
hdavid16                         2           609              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Hector Perez                609          100.0          0.0               11.17
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00117/jcon.00117/10.21105.jcon.00117.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00117/jcon.00117/10.21105.jcon.00117.pdf) :page_facing_up: :point_left:

**Comment by matbesancon** [comment]
@sbolusani @joaquimg the reviewer's guide is here: https://juliacon.github.io/proceedings-guide/reviewer/

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1287/ijoc.2015.0669 is OK
- 10.1007/978-0-387-74759-0_348 is OK
- 10.1007/978-1-4613-0215-5_8 is OK
- 10.1002/9780470400531.eorms0537 is OK
- 10.1007/11527695_15 is OK
- 10.1137/15m1020575 is OK
- 10.1137/15M1020575 is OK
- 10.3390/pr7110839 is OK
- 10.1023/a:1025154322278 is OK
- 10.1007/s10589-020-00176-0 is OK
- 10.1145/3511528.3511535 is OK

MISSING DOIs

- 10.1007/978-3-642-11503-5_12 may be a valid DOI for title: Automating mathematical program transformations
- 10.1007/s12532-017-0130-5 may be a valid DOI for title: Parallelizing the dual revised simplex method
- 10.1016/s0098-1354(98)00293-2 may be a valid DOI for title: LOGMIP: a disjunctive 0–1 non-linear optimizer for process system models
- 10.1007/s12532-011-0026-8 may be a valid DOI for title: Pyomo: modeling and solving mathematical programs in Python

INVALID DOIs

- https://doi.org/10.1016/j.compchemeng.2014.03.014 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/B978-0-323-95879-0.50141-7 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.compchemeng.2021.107616 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.compchemeng.2022.107856 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.cherd.2022.08.027 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1002/aic.14088 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.compchemeng.2015.02.013 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/S0098-1354(00)00581-0 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.compchemeng.2005.04.004 is INVALID because of 'https://doi.org/' prefix
```

**Comment by hdavid16** [comment]
> @hdavid16 can you check the DOIs listed above, that's minor but can be done in parallel

DOIs have been updated in the repo now.

**Comment by whedon** [comment]
:wave: @sbolusani, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @joaquimg, please update us on how your review is going (this is an automated reminder).

**Comment by hdavid16** [comment]
@sbolusani @joaquimg, this is a friendly reminder to please review. Thank you.

**Comment by joaquimg** [comment]
Sorry for the delay, I am working on it.

**Comment by hdavid16** [comment]
Hi @sbolusani and @joaquimg, I'm sure you are all super busy. But just want to remind you to review this conference paper. It's been almost a year since the last JuliaCon where this work was presented...

**Comment by matbesancon** [comment]
I just reminded the two reviewers

**Comment by matbesancon** [comment]
@whedon add @joaquimg as a reviewer

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by matbesancon** [comment]
@whedon add @joaquimg as reviewer

**Comment by whedon** [comment]
OK, @joaquimg is now a reviewer

**Comment by joaquimg** [comment]
From a first pass:

> Installation: Does installation proceed as outlined in the documentation?

On Julia 1.9.0, no a clean enviroment:
```julia
julia> using Pkg

julia> Pkg.add("DisjunctiveProgramming")
```
Installs, but returns warnings:

```
        Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading. To see why use `status --outdated -m`
Precompiling project...
  45 dependencies successfully precompiled in 482 seconds. 94 already precompiled.
  3 dependencies had warnings during precompilation:
┌ Symbolics [0c5d862f-8b57-4792-8d23-62f2024744c7]
│  WARNING: Method definition isapprox(IntervalSets.AbstractInterval{T} where T, IntervalSets.AbstractInterval{T} where T) in module IntervalSets at C:\JG\Julia\packages\IntervalSets\6RTOk\src\IntervalSets.jl:144 overwritten in module DomainSets at C:\JG\Julia\packages\DomainSets\aafhp\src\domains\interval.jl:52.
│    ** incremental compilation may be fatally broken for this module **
│
│  WARNING: Method definition isapprox(IntervalSets.AbstractInterval{T} where T, IntervalSets.AbstractInterval{T} where T) in module IntervalSets at C:\JG\Julia\packages\IntervalSets\6RTOk\src\IntervalSets.jl:144 overwritten in module DomainSets at C:\JG\Julia\packages\DomainSets\aafhp\src\domains\interval.jl:52.
│    ** incremental compilation may be fatally broken for this module **
└
┌ DisjunctiveProgramming [0d27d021-0159-4c7d-b4a7-9ccb5d9366cf]
│  WARNING: Method definition isapprox(IntervalSets.AbstractInterval{T} where T, IntervalSets.AbstractInterval{T} where T) in module IntervalSets at C:\JG\Julia\packages\IntervalSets\6RTOk\src\IntervalSets.jl:144 overwritten in module DomainSets at C:\JG\Julia\packages\DomainSets\aafhp\src\domains\interval.jl:52.
│    ** incremental compilation may be fatally broken for this module **
└
┌ DomainSets [5b8099bc-c8ec-5219-889f-1d9e522a28bf]
│  WARNING: Method definition isapprox(IntervalSets.AbstractInterval{T} where T, IntervalSets.AbstractInterval{T} where T) in module IntervalSets at C:\JG\Julia\packages\IntervalSets\6RTOk\src\IntervalSets.jl:144 overwritten in module DomainSets at C:\JG\Julia\packages\DomainSets\aafhp\src\domains\interval.jl:52.
│    ** incremental compilation may be fatally broken for this module **
└
```

> Example usage: Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).

The example in the README could be end-to-end and contain a solver, a call to `optimize` and some expected results.

Also, the line `@proposition(m, y[1] ∨ y[2])` is failing on windows.

> Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

I could not find docs other than the README. There is a docs folder, but no link in the README.
Also, there are broken links in the README.

> Automated tests: Are there automated tests or manual steps described so that the function of the software can be verified?

I could not find a CI.yml with the automated tests.

> Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

No.

> Authors: Does the paper.tex file include a list of authors with their affiliations?

Only the last author has an email. He is supposed to be the corresponding author?

> References: Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

Many references are missing DOI´s.

**Comment by joaquimg** [comment]
I recommend tha the authors go through the reviewer check list on their own, to avoid multiple iterations.

**Comment by joaquimg** [comment]
@matbesancon , about:
> Page limit: Is the page limit for full papers respected by the submitted document?

what is the page limit? I could not find it easily. Sorry, it is probably obvious.

**Comment by joaquimg** [comment]
@hdavid16 there is work for you :)

**Comment by matbesancon** [comment]
yes page limit is respected here :)

**Comment by joaquimg** [comment]
Also for @matbesancon ,
Per the reviewer guidelines (https://joss.readthedocs.io/en/latest/review_checklist.html#software-paper) and this comment (https://github.com/JuliaCon/proceedings-review/issues/117#issuecomment-1345551688) a section named "Statement of need" seems necessary. Can you confirm? If so, @hdavid16 will have to make it explicit.

**Comment by joaquimg** [comment]
> Content

This seems to be the best place to comment on paper presentation.

I have a few main points:

1) Models in section 2.2 are confusing because they are not self-contained. They are simply groups of formulas that show up as needed, and the text dictates how to combine them. Although it is possible to understand, it could be more straightforward.

For instance, instead of having a paragraph:
"... this model is given by (13), (16)-(19)..." and only show the additional (16)-(19)
one could do:
"... this model is given by (16)-(20)..." and show (16)-(20) where (16) repeats (13) or adds it explicitly:
```
(13)                      (16)
new eq...                 (17)
new eq...                 (18)
new eq...                 (19)
new eq...                 (20)
```

2) In "Note: any negated Boolean variables, ¬Yi, are replaced with 1 − yi in the reformulation"
It is not clear if should be a term in the sum (this seems wrong) or if there would be and additional variable zi = 1 − yi (this looks good).

3) sections 2.4.2 and 2.4.3 seem to be not very related to the remainder of the paper.
They are not mentioned as future work. So they might be better as one sentence comment in a section the focuses more on  2.4.1 which is the relevant piece for the paper.

4) From paper and readme it is not clear what is the difference between `@disjunction` or `add_disjunction!`.
From the example, it is not clear if the chosen usage was necessary to define the problem or if it was an arbitrary choice.

5) reference [11] seems off.

6) fix JuMP capitalization in references [9] and [10]

**Comment by hdavid16** [comment]
@joaquimg thank you for your review.

The package has just gone a complete rewrite and has now been released (v0.4.1).  With this new release:

- Installing should work on 1.9 without any errors. 
- Docs and CI have also been added.
- DOIs have been fixed/added.
- Community guidelines added.
- Full example with `optimize!` call added to README and docs.

Regarding your main points:

1. Examples are self-contained now as you suggested.
2. Conversion of logical constraints to algebraic inequalities has been written in a more clear way.
3. Section 2.4 is referenced in future work.
4. Rewrite avoids ambiguous usage for nested disjunctions.
5. Reference fixed.
6. Capitalization in JuMP reference fixed.

I look forward to your feedback.

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by matbesancon** [comment]
@whedon check references

**Comment by matbesancon** [comment]
@hdavid16 while we are waiting for confirmation from @joaquimg that the changes are satisfactory, one step to do is archiving on Zenodo the current version of the software and giving us the corresponding DOI that gets generated

**Comment by hdavid16** [comment]
@matbesancon here it is: 10.5281/zenodo.10144853

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by matbesancon** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by matbesancon** [comment]
@whedon set 10.5281/zenodo.10144853 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.10144853" target="_blank">10.5281/zenodo.10144853</a> is the archive.

**Comment by matbesancon** [comment]
@whedon set v0.4.1 as version

**Comment by whedon** [comment]
OK. v0.4.1 is the version.

**Comment by matbesancon** [comment]
@hdavid16 for stylistic reasons, can we replace the successive equations by an `align` block? This will look much nicer

**Comment by matbesancon** [comment]
same for all optimization problems, prefer a template like:
```
\begin{align*}
\min_{x}\,\, & f(x) \\
\text{s.t. }\, & g(x) \leq 0 \\
& x_i \in X && \forall i \in .. \\
& x \in \mathcal{X}
\end{align*}
```

**Comment by joaquimg** [comment]
I do re-iterate @matbesancon comment.

Other than that, good to go on my side. 

Great work and congratulations for the nice package!

**Comment by hdavid16** [comment]
@joaquimg and @matbesancon the paper has been updated on `master` as you requested (replacing all equations/expressions with `align*` blocks).

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00117/jcon.00117/10.21105.jcon.00117.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00117/jcon.00117/10.21105.jcon.00117.pdf) :page_facing_up: :point_left:

**Comment by matbesancon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/978-3-642-11503-5_12 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1007/978-3-030-00148-3 is OK
- 10.1016/s0098-1354(98)00293-2 is OK
- 10.1287/ijoc.2015.0669 is OK
- 10.1007/978-0-387-74759-0_348 is OK
- 10.1016/j.compchemeng.2014.03.014 is OK
- 10.1007/978-1-4613-0215-5_8 is OK
- 10.1007/978-3-030-68928-5 is OK
- 10.1007/s12532-011-0026-8 is OK
- 10.1016/B978-0-323-95879-0.50141-7 is OK
- 10.1016/j.compchemeng.2021.107616 is OK
- 10.1016/j.compchemeng.2022.107856 is OK
- 10.1016/j.cherd.2022.08.027 is OK
- 10.1002/9780470400531.eorms0537 is OK
- 10.1007/11527695_15 is OK
- 10.1137/15M1020575 is OK
- 10.1137/15M1020575 is OK
- 10.3390/pr7110839 is OK
- 10.1002/aic.14088 is OK
- 10.1007/978-3-030-00148-3 is OK
- 10.1016/j.compchemeng.2015.02.013 is OK
- 10.1016/S0098-1354(00)00581-0 is OK
- 10.1002/9781118627372 is OK
- 10.1023/a:1025154322278 is OK
- 10.1007/s10589-020-00176-0 is OK
- 10.1145/3511528.3511535 is OK
- 10.1016/j.compchemeng.2005.04.004 is OK
- 10.48550/arXiv.2202.05198 is OK
- 10.48550/arXiv.1510.01791 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @JuliaCon/jcon-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/74

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/74, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by matbesancon** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by matbesancon** [comment]
@whedon remove @sbolusani  as reviewer

**Comment by matbesancon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JCON!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/JuliaCon/proceedings-papers/pull/75
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jcon.00117](https://doi.org/10.21105/jcon.00117)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by matbesancon** [comment]
congratulations @hdavid16 !

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00117/status.svg)](https://doi.org/10.21105/jcon.00117)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jcon.00117">
  <img src="https://proceedings.juliacon.org/papers/10.21105/jcon.00117/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://proceedings.juliacon.org/papers/10.21105/jcon.00117/status.svg
   :target: https://doi.org/10.21105/jcon.00117
```
This is how it will look in your documentation:

[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00117/status.svg)](https://doi.org/10.21105/jcon.00117)

**We need your help!**

JuliaCon Proceedings is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://forms.gle/yANMe4c9YmuW9MWcA
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-jcon
